packages/storage/amplify_storage_s3_dart/lib/src/amplify_storage_s3_dart_impl.dart

```diff
@@ -69,19 +111,46 @@ class AmplifyStorageS3Dart extends StoragePluginInterface {
       identityProvider: identityProvider,
     );
 
-    // TODO(HuiSF): create S3Client instance with
-    //  * credentialsProvider
-    //  * _prefixResolver
-    //  * other fields
-    // final credentialsProvider = authProviderRepo
-    //     .getAuthProvider(APIAuthorizationType.iam.authProviderToken);
+    final credentialsProvider = authProviderRepo
+        .getAuthProvider(APIAuthorizationType.iam.authProviderToken);
```

Comment:
Seems like there might be a race condition between plugins - what if Storage configures before Auth?

Auth providers should probably be registered in `addPlugin` so that they're available to get in configure.

Reply:
Hum good catch, I saw once or twice getting token call failed while running the example App, the race condition may be the cause. I'll see how we can improve.

Reply:
I'll keep investigate this and probably fix in a separate PR.